python/ribasim/tests/test_model.py

```diff
@@ -143,3 +144,19 @@ def test_write_adds_fid_in_tables(basic, tmp_path):
         assert "fid" in df.columns
         fids = df.get("fid")
         assert fids.equals(pd.Series(range(1, len(fids) + 1)))
+
+
+def test_model_merging(basic, subnetwork, tmp_path):
+    model = deepcopy(basic)
+    model_added = deepcopy(subnetwork)
+    model.merge_model(model_added)
+    model.merge_model(model_added)
+    assert (model.network.node.df.index == range(1, 44)).all()
+    assert model.max_allocation_network_id() == 2
+    for node_type, node_added in model_added.nodes().items():
```

Comment:
Regarding this loop: I tried comparing models directly, e.g.
```python
assert model_added == subnetwork
```

but I kept running into 'maximum recursion depth reached' errors, and I could not find out why; there is no loop in the model object as far as I could find.

Reply:
It might be that a comparison somewhere (especially given that you implement `__eq__` yourself) is calling itself again.

Reply:
https://github.com/Deltares/Ribasim/issues/928